npm: fix idempotence

pari- opened the pull request:
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
npm module

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file =
  configured module search path = Default w/o overrides
```

##### SUMMARY
There's basically only one variant of where calling npm with "--production" and without scope/package (which is what Ansible does) is useful: installing a module  

Right now, the "npm-handling"-code trivially appends all given attributes to the cmd, no matter what. In the case of listing modules without scope/package this leads to a fundamental difference of outcome, since it breaks idempotence, though.

Let's take the following minimal example:

```
---
- hosts: all

  tasks:
    - name: install pm2
      npm:
        name: "pm2"
        production: "yes"
        global: "yes"
        version: "2.4.2"
```

This snippet does time and again install 'pm2' when being run, simply because it is going to be evaluated to:

`npm list --json --global --production` which always results correctly in `{}` (which in turn, the actual package name becomes appended to within the module's logic, and thus 'missing' will always be 'pm2').

This PR provides a rather dirty fix on this, though.

pari- committed:
npm: fix idempotence

pari- commented:
Hi,

any news about this? Since the module is a community contribution - what is the proper (as in: 'NonMaintainerUpload'-way) to get this issue fixed?

calfonso commented:
@pari- are you able to get @chrishoffman or a user of this module to verify it works as intended? Once they comment on this, we'll finish the review and get it merged.

samdoran committed:
Better idempotency fix

More intelligently add --production rather than depending on hard coded order in args list
Cleanup boilderplate imports and license
PEP8 fixes
